### PR TITLE
ACAS-387: LD Chem config check, config fix and health check

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
  -  For BBChem implementation refactored to just a single URL with paths built into the code
  - `checkStandardizerSettings` chemstructure service interface and implementations to pass the applied standardization settings and system configured standardization settings and determine if standardization is needed
    - Keep the jchem service implementation the same (just compare hashes)
    - No check for indigo as standardization is not implmented
    - For BBChem
      - Call service to fix configs (check they are valid)
      - Call service to compare old (fixed configs) and new (fixed configs)
         - Add the Preprocessor and Suite version to the saved settings to be able to pass to the config check service.  Did this by adding a call to the health endpoint.
 - Refactor image service function to be in BBChemservice controller

## Related Issue
ACAS-387

## How Has This Been Tested?
Build all 3 indigo, jchem and bbchem implementations

### CODE REVIEW

File: src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java
Comment (bffrost) [reply src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java:70]: That's great. We'll want to update the default k8s configs and the config upgrader script once we merge this, right?

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java:70]: Yes, thanks, definitely need to set it on K8s values. I JUST sent you a seperate PR for this.

File: src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:145]: This comment block is useful, but I think it should be expanded a bit.
Specifically, explaining what will happen in this case 1, since it's not part of the `if` statement below. I'm thinking something like:
```
// In case 1, we pass along an empty StandardizationHistory, which will cause the system to report that it needs standardization.
// In case 2 we try to fill in an initial standardization record with the current settings, provided they're valid settings, so no restandardization is needed.
```

File: src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:176]: I think it is worth explaining here what a "StandardizationSettings" is. A lot of these variables and classes have similar names. As I understand it `StandardizationSettings` is actually the current state of the system, specifically whether it needs restandardization and why. `StandardizationStatus` might be a better name, or `CurrentStandardizationStatus`

**Review by bffrost — CHANGES_REQUESTED** [review]
Looks good. This is a large set of changes but I'm able to follow it, and the refactor has gone a long way to simplifying this stuff. Spotted one attribute that I think is unused, and left some comments about code comments and a confusing class name. Should be good to go once these are addressed.

File: src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java:662]: Deprecation notice: https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.13.2.2/com/fasterxml/jackson/databind/node/ObjectNode.html#put-java.lang.String-com.fasterxml.jackson.databind.JsonNode-

Says to use set or replace:
https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.13.2.2/com/fasterxml/jackson/databind/node/ObjectNode.html#set-java.lang.String-com.fasterxml.jackson.databind.JsonNode-
https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.13.2.2/com/fasterxml/jackson/databind/node/ObjectNode.html#replace-java.lang.String-com.fasterxml.jackson.databind.JsonNode-

Only difference is that set returns the node and replace returns the old value but we aren't using the return values in either case.

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java:662]: I didn't replace all deprecated uses but I will do that now.

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java:662]: Done

File: src/main/java/com/labsynch/labseer/domain/StandardizationHistory.java
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/domain/StandardizationHistory.java:79]: Thanks this was a mistake, didn't mean to add it to this class. Good catch.

File: src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:145]: Thanks, added those

File: src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:179]: Makes sense but unclear if you are suggesting that I refactor the table name, class name and variable names to match `CurrentStandardizationStatus ` as part of this PR or to just note it in this comment.

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:179]: I added some comments at least for now.

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:179]: I realize after writing this that the way to refactor this is to just drop the settings table and make settings just a DTO because there is no point in saving the row in a table if we are completely replacing it on every check.

Comment (bffrost) [reply src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java:179]: @brianbolt your last comment makes a lot of sense to me. I think it's worth doing as part of this PR, just to leave this code in a clean place. My hope is that we don't have a lot more work to do in this area of the code, so there may not be another good opportunity to make that change.

**Review by bffrost — APPROVED** [review]
LGTM, nice simplification in your latest refactor.

**Pull request merged**
